src/data/cassandra/impl/ExecutionStrategy.hpp

```diff
@@ -213,6 +228,24 @@ class DefaultExecutionStrategy {
         );
     }
 
+    /**
+     * @brief Non-blocking  query execution used for writing data. Constrast with write, this method does not execute
+     * the statements in a batch.
+     *
+     * Retries forever with retry policy specified by @ref AsyncExecutor.
+     *
+     * @param statements Vector of statements to execute
+     * @throw DatabaseTimeout on timeout
+     */
+    void
+    writeEach(std::vector<StatementType>&& statements)
+    {
+        if (statements.empty())
+            return;
```

Comment:
Probably `for_each` will not do anything if the vector is empty so this is not needed.